Sources/Commands/SwiftTestTool.swift

```diff
@@ -188,6 +199,86 @@ public struct SwiftTestTool {
         let result: Void? = try? system(args, environment: NSProcessInfo.processInfo().environment)
         return result != nil
     }
+
+    /// Runs the corresponding tool to get the tests JSON.
+    /// On OSX, we use the swiftpm-xctest-helper tool bundled with swiftpm.
+    /// On Linux, XCTest can dump the json using `--dump-tests-json` mode.
+    ///
+    /// - Parameters:
+    ///     - path: Path to the XCTest bundle(OSX) or executable(Linux).
+    ///     - tempDirectory: Path to a directory where a temporary output
+    ///                      file will be created containing the JSON.
+    ///                      The File is deleted when this method returns.
+    private func listTests(path: String, tempDirectory: String) throws {
+        // Make Sure tests are present.
+        guard path.isValidTest else { throw TestError.testsExecutableNotFound }
+
+        // Get path to the XCTestHelper tool.
+        let xctestHelperTool = Path.join(Process.arguments.first!.abspath.parentDirectory, "swiftpm-xctest-helper")
+
+        // Dump the output in a temp file inside the build folder.
+        let outputFile = Path.join(tempDirectory, "list-test-output.txt")
+        // Remove the file before we exit this method.
+        defer { let _ = try? Utility.removeFileTree(outputFile) }
+
+        // Run the correct tool.
+      #if os(OSX)
+        let args = [xctestHelperTool, path, outputFile]
+        try system(args, environment: ["DYLD_FRAMEWORK_PATH": try platformFrameworksPath()])
+      #else
+        let arg = [path, "--dump-tests-json", ">", outputFile].joined(separator: " ")
+        try system("/bin/sh", "-c", arg)
+      #endif
+
+        // Read the output file and parse the contents.
+        try fopen(outputFile) { handle in
+            let data = try handle.enumerate().joined(separator: "\n")
+            try self.parseAndListTest(JSONString: data)
+        }
+    }
+
+    /// Parse the JSON String containing the tests and print the output.
+    ///
+    /// - Parameters:
+    ///     - JSONString: The JSON as string.
+    private func parseAndListTest(JSONString: String) throws {
```

Comment:
I think variable names should be lower case, and probably not include the top. I would just use `json` here.